nixos/modules/services/misc/paperless.nix

```
@@ -32,11 +32,17 @@ let
     else toString s
   ) cfg.settings);
 
-  manage = pkgs.writeShellScript "manage" ''
+  manage = pkgs.writeShellScriptBin "paperless-manage" ''
     set -o allexport # Export the following env vars
     ${lib.toShellVars env}
     ${lib.optionalString (cfg.environmentFile != null) "source ${cfg.environmentFile}"}
-    exec ${cfg.package}/bin/paperless-ngx "$@"
+
+    cd ${cfg.dataDir}
+    sudo=exec
+    if [[ "$USER" != ${cfg.user} ]]; then
+      ${if config.security.sudo.enable then "sudo='exec /run/wrappers/bin/sudo -u ${cfg.user} -E'" else "echo 'Aborting, paperless-manage must be run as ${cfg.user}!'; exit 2"}
```

```suggestion
      ${if config.security.sudo.enable
        then "sudo='exec /run/wrappers/bin/sudo -u ${cfg.user} -E'"
        else ">&2 echo 'Aborting, paperless-manage must be run as user `${cfg.user}`!'; exit 2"}
```
- Break up overlong line
- Print to stderr
- `must be run as` -> `must be run as user`

---

Unrelated to comment above: Why don't we use `systemd-run`/`run0` here? That is basically guaranteed to be available on every NixOS system.

---

> Why don't we use systemd-run/run0 here? That is basically guaranteed to be available on every NixOS system.

Lets move this to another PR to keep the scope of this manageable. If this is true and works universally, we should migrate all similar wrappers to them.